### PR TITLE
refactor(connector): use config file to initiate a debezium source connector

### PR DESCRIPTION
**Pull request by StrikeW**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Refactor the connector node to load a `.properties` file to initiate a Debzium source connector.
- `debezium.properties`: common configs shared by all connectors
- `mysql.properties`: config for the MySQL connector
- `postgres.properties`: config for the PG connector

We can easily implement a new connector by adding a new config file and the required maven dependencies to the project.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [x] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

### CODE REVIEW

File: java/connector-node/risingwave-connector-service/src/main/resources/postgres.properties [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# The name of the PostgreSQL replication slot
+slot.name=${slot.name}
+
+# default plugin name is 'pgoutput'
+plugin.name=${debezium.plugin.name:-pgoutput}
+
+# allow to auto create publication for given tables
+publication.autocreate.mode=${debezium.publication.autocreate.mode:-filtered}
+
+# default hearbeat interval 5 mins
```
Comment (liurenjie1024) [inline java/connector-node/risingwave-connector-service/src/main/resources/postgres.properties:25]:
```suggestion
# default heartbeat interval 5 mins
```

File: java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/sourcenode/common/DbzConnectorConfig.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                        ConfigurableOffsetBackingStore.OFFSET_STATE_VALUE, startOffset);
+            }
+
+            dbzProps.putAll(mysqlProps);
+        } else if (source == SourceTypeE.POSTGRES) {
+            var postgresProps = initiateDbConfig(POSTGRES_CONFIG_FILE, substitutor);
+
+            // if offset is specified, we will continue reading changes from the specified offset
+            if (null != startOffset && !startOffset.isBlank()) {
+                postgresProps.setProperty("snapshot.mode", "never");
```
Comment (liurenjie1024) [inline java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/sourcenode/common/DbzConnectorConfig.java:105]: Why not also put this in config file?

**Review by liurenjie1024 — APPROVED** [review]
LGTM

**Review by tabVersion — COMMENTED** [review]
if `property` is a widely accepted config file format, I would agree with the change

**Comment by liurenjie1024** [comment]
> if `property` is a widely accepted config file format, I would agree with the change

It's quite popular.

File: java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/sourcenode/common/DbzConnectorConfig.java (hunk 2 above)
Comment (StrikeW) [reply java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/sourcenode/common/DbzConnectorConfig.java:105]: There is a default config for `snapshot.mode` in config file, but we need to disable snapshotting when `startOffset` is set (e.g. recovery scenario). This logic cannot be achieved by the config file.

**Pull request merged**